notebooks.go

```diff
@@ -363,6 +368,88 @@ func (s *server) GetNotebooks(w http.ResponseWriter, r *http.Request) {
 	s.respond(w, r, resp)
 }
 
+func (s *server) GetDefaultNotebook(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+
+	log.Printf("loading notebooks for %q", namespace)
+
+	notebooks, err := s.listers.notebooks.Notebooks(namespace).List(labels.Everything())
+	if err != nil {
+		s.error(w, r, err)
+		return
+	}
+
+	resp := &notebookapiresponse{
+		APIResponseBase: APIResponseBase{
+			Success: false,
+			Status:  http.StatusOK,
+		},
+	}
+
+	for _, notebook := range notebooks {
+
+		if val, ok := notebook.Labels["notebook.statcan.gc.ca/default-notebook=true"]; ok {
+			if val == "true" {
+
+			}
+			// Load events
+			allevents, err := s.listers.events.Events(notebook.Namespace).List(labels.Everything())
+			if err != nil {
+				log.Printf("failed to load events for %s/%s: %v", notebook.Namespace, notebook.Name, err)
+			}
+
+			// Filter past events
+			events := make([]*corev1.Event, 0)
+			for _, event := range allevents {
+				if event.InvolvedObject.Kind != "Notebook" || event.InvolvedObject.Name != notebook.Name || event.CreationTimestamp.Before(&notebook.CreationTimestamp) {
+					continue
+				}
+
+				events = append(events, event)
+			}
+			sort.Sort(eventsByTimestamp(events))
+
+			imageparts := strings.SplitAfter(notebook.Spec.Template.Spec.Containers[0].Image, "/")
+
+			// Process current status + reason
+			status := processStatus(notebook, events)
+
+			volumes := []string{}
+			for _, vol := range notebook.Spec.Template.Spec.Volumes {
+				volumes = append(volumes, vol.Name)
+			}
+
+			cpulimit := resource.Zero.AsDec()
+			if req, ok := notebook.Spec.Template.Spec.Containers[0].Resources.Requests[corev1.ResourceCPU]; ok {
+				cpulimit = req.AsDec()
+			}
+			resp.APIResponseBase.Success = true
+
+			resp.Notebook = notebookresponse{
+				Age:          notebook.CreationTimestamp.Time,
+				Name:         notebook.Name,
+				Namespace:    notebook.Namespace,
+				Image:        notebook.Spec.Template.Spec.Containers[0].Image,
+				LastActivity: notebook.Annotations[LastActivityAnnotation],
+				ServerType:   notebook.Annotations[ServerTypeAnnotation],
+				ShortImage:   imageparts[len(imageparts)-1],
+				CPU:          cpulimit,
+				GPUs:         s.processGPUs(notebook),
+				Memory:       notebook.Spec.Template.Spec.Containers[0].Resources.Requests[corev1.ResourceMemory],
+				Status:       status,
+				Volumes:      volumes,
+				Labels:       notebook.Labels,
+				Metadata:     notebook.ObjectMeta,
+			}
+			s.respond(w, r, resp)
+			return
+		}
+	}
+
+	s.respond(w, r, resp)
```

Comment:
you have two responds for one function which can cause confusions and can easily be avoided. For something like this, I would suggest doing something more like:
```
for each notebook{
    if default notebook{
        default notebook found
        break;
   }
}
process notebook if found;
return response;
```

Reply:
good idea